### PR TITLE
bugfix: Fail FW Bounty missions upon choosing a side

### LINKED ISSUE #2544: Joining the Free Worlds should fail FW Bounty missions
https://github.com/endless-sky/endless-sky/blob/master/data/free%20worlds%20intro.txt#L1579
> As you walk through the spaceport, a woman in a militia uniform approaches you. "Are you Captain `<last>`?" she asks. You nod. "I'm the commander of the local militia," she says. "There's a pirate fleet that has attacked freighters in this sector several times in the last few months, but now it seems they've moved outside of Free space. I'd like to strike at them before they return and do more damage here, but right now the Free Worlds Council won't allow militia ships to be used for preemptive attacks."

And then we join the Free Worlds.

[Alpha Cygni.txt](https://github.com/endless-sky/endless-sky/files/1020557/Alpha.Cygni.txt)
Mission is still present.

### PR DESCRIPTION
**Pull request by tehhowch**
As noted in the FW Bounty mission text, you are not supposed to do these missions as a member of the FW militia.
Credit to @Bladewood for spotting the inconsistency.
Closes #2544

### CODE REVIEW

**Comment by Bladewood** [comment]
Maybe there should be a message explaining to the player why the missions have been failed? (in-universe, of course)

**Comment by tehhowch** [comment]
can you `dialog` the `on fail`? If so, then sure, easy enough. If not, a description change could perhaps keep the player appraised, but IIRC mission descriptions are supposed to be as short as reasonably achievable.

**Comment by Bladewood** [comment]
I think we can go with something a little more elegant, like literal dialogue.
> You receive a call from the Free Worlds militia member who offered you the mission to hunt the `<npc>`. "I'm told by my sources that you've joined us, and that you've personally met with the Council. If that is really the case, then I'm afraid that we'll have to revoke our earlier request."

**Comment by jafdy** [comment]
Something a little more thankful might be better.
>They probably have more important things for you to do, I'll reassign the mission to someone else.

You will also need to put this in a different mission to prevent the dialog coming up if you fail it for some other reason.

**Comment by tehhowch** [comment]
@jafdy The only ways this mission can fail are
 - you die in combat
 - you join a side

I've set up a `branch` based on current mission markers, but the conditions may need investigating whenever the Syndicate and Republic storylines are finalized.

If the player makes it 9 jumps from Bourne to Clink after choosing to join FW, the `on fail` conversation appears before the "Rescue Katya 1" conversation.

File: data/free worlds intro.txt [hunk 1]
```diff
@@ -1602,7 +1602,29 @@ mission "FW Bounty 1"
...
+			branch "Free Worlds"
+				or
+					has "Rescue Katya 1: active"
+					has "joined the free worlds"
+				
+			`You receive a sealed message from a courier who bears a Free Worlds insignia. You open it and read:`
+			``
+			`Captain <last>:`
+			`	Since you have decided to actively oppose the Free Worlds and our independence, we must retract the reward being offered to hunt the "Rat Pack" pirates. It disappoints us that you have chosen this path, but as members of the Free Worlds, we must respect your decision as a free individual.`
+			`	However, let this message also serve as a notification: do not enter Free Worlds territory without permission, or you will be fired upon.`
```
Comment (jafdy) [inline data/free worlds intro.txt]: If you join a different faction then you first mission to free world space might be a trade mission.

Comment (tehhowch) [inline data/free worlds intro.txt]: Very true. But with 0 details and 0 timeline for Republic/Syndicate/Other faction plots, all I have to work with is "there is conflict at the time of faction alignment."

**Comment by endless-sky** [comment]
Joining the Free Worlds is not the same thing as joining the militia, so this doesn't need to be changed.

**Pull request closed without merging**